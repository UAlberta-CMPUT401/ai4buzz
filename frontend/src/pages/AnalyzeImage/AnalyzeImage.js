import React from 'react';

const AnalyzeImage = () => {
  return <div>AnalyzeImage</div>;
};

export default AnalyzeImage;
