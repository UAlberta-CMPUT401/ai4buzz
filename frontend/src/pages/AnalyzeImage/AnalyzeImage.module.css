.analyzeImagePage {
  display: flex;
  justify-content: center;
  padding-top: 15rem;
}

.images {
  padding-bottom: 1rem;
}

.fileName {
  color: #888e99;
}

input[type='file'] {
  font-size: 1.1rem;
  padding-bottom: 1rem;
}

.uploadImagesButton,
#file-upload-button {
  font-size: 1.1rem;
  padding: 0.4rem 1.5rem;
  border-radius: 0.5rem;
  cursor: pointer;
  color: #253251;
  border: none;
  width: 100%;
}
