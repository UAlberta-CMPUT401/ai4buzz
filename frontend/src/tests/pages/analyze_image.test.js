import React from 'react';
import renderer from 'react-test-renderer';
import AnalyzeImage from '../../pages/AnalyzeImage/AnalyzeImage';
import { BrowserRouter } from 'react-router-dom';

it('Renders Analyze Image page correctly', () => {
  Object.defineProperty(window.document, 'cookie', {
    writable: true,
    value: 'access_token=FAKE_TOKEN_VAL',
  });

  const files = [
    new File(['image-bits...'], 'chucknorris.png', {
      type: 'image/png',
    }),
  ];

  const tree = renderer
    .create(
      <BrowserRouter>
        <AnalyzeImage files={files} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
