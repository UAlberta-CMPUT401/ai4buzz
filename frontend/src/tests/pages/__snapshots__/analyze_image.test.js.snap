// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Analyze Image page correctly 1`] = `
<div
  className="analyzeImagePage"
>
  <form>
    <div>
      <h1>
        Upload Images
      </h1>
      <div
        className="error"
      />
    </div>
    <div
      className="images"
    >
      <h3>
        Images
      </h3>
      <div
        className="fileName"
      >
        chucknorris.png
      </div>
    </div>
    <input
      className="chooseFilesInput"
      multiple={true}
      onChange={[Function]}
      type="file"
    />
    <div
      className="buttonGradient"
    >
      <button
        className="uploadImagesButton"
        onClick={[Function]}
      >
        Analyze
      </button>
    </div>
  </form>
</div>
`;
